### PR TITLE
Add new ERC20 Lesson

### PR DESCRIPTION
**Pull request by 7i7o**
## Changes

- [x] Add Lesson 6 (will update later when we have a full DeFi track)
    - "Create your own ERC20 Token with Foundry"
- [x]  Add images required for new lesson
- [x] Add icon for Foundry tag on Lesson List
- [x] change all Goerli references to Sepolia

### CODE REVIEW

File: pages/lessons/projects/6.mdx [hunk 1]
```diff
@@ -0,0 +1,796 @@
...
+
+The Ethereum blockchain (and most EVM-compatible blockchains) has a native token
+that is used to pay for transaction costs, and also to reward validators in
+Proof of Stake, or previously miners in Proof of Work. In Ethereum that native
+token is Ether (or ETH), in Polygon its MATIC, and so on.
+
+Let's focus on the Ethereum blockchain for the sake of clarity.
+
+Every _other_ Token or "Crypto" you see, use or interact with in Ethereum is
+basically a smart contract that lets you send this tokens, receive them and
```
Comment (georgemac510) [inline pages/lessons/projects/6.mdx:36]: "lets you send this tokens," should be "lets you send these tokens"

**Comment by elPiablo** [comment]
Nice to see this land in Github @7i7o !!!

File: pages/lessons/projects/6.mdx [hunk 2]
```diff
@@ -0,0 +1,796 @@
...
+[ERC-20: Token Standard](https://eips.ethereum.org/EIPS/eip-20#token)
+
+</SideDrawer>
+
+## Has someone created a battle-tested library for the standard?
+
+Yes. OpenZeppelin (among others) has created an implementation we can inherit to
+create
+our own tokens easily.
+
```
Comment (georgemac510) [inline pages/lessons/projects/6.mdx:100]: Might a deeper conversation be made here about what "inherit" means in this context. What functionality is extracted from the Open Zeppelin contracts? @7i7o I know we might have covered this in Fundamentals - Open Zeppelin, but I just feel that learners may have a lingering question of "What benefit is coming from the Open Zeppelin contracts? Why is this so important?".

File: pages/lessons/projects/6.mdx [hunk 3]
```diff
@@ -0,0 +1,796 @@
...
+    * 5 - Final touches
+
+Before we build, we need a stable, leveled surface to place our foundations so
+we can build on top.
+
+The tools for coding and the project's file/folder structure will be our
+foundations.
+
+## Setting up Foundations
+
```
Comment (georgemac510) [inline pages/lessons/projects/6.mdx:122]: Not sure if this is necessary, but just a suggestion. Would it be helpful to mention the version of Node.js used? I got through the entire lesson successfully using Node 19.4.0. Not sure if an older version such as 8.0.0 would yield the same result.

File: pages/lessons/projects/6.mdx [hunk 4]
```diff
@@ -0,0 +1,796 @@
...
+forge remappings > remappings.txt
+echo "@openzeppelin/=lib/openzeppelin-contracts/" >> remappings.txt
+```
+
+This tells Forge, “Hey, anytime you hit `@openzepplin/`, look in
+`lib/openzeppelin-contracts/` instead.”
+
+Now that we have an overview of the tools and the OpenZeppelin's libraries
+installed, we can clean up the files we won't use and start fresh:
+
```
Comment (georgemac510) [inline pages/lessons/projects/6.mdx:286]: Maybe clarify that the 3 commands below can be done in one shot or specify that they might be done separately?

File: pages/lessons/projects/6.mdx [hunk 5]
```diff
@@ -0,0 +1,796 @@
...
+## Lock the Door!
+
+In our present state, our functions for minting and burning can be called by any
+address.
+
+That's ok for `burn` since people can only burn their own tokens (or their
+approved allowances), but we don't want anyone to `mint` tokens and increase the
+total supply of our new token.
+
```
Comment (georgemac510) [inline pages/lessons/projects/6.mdx:442]: Great point and important to not here!

File: pages/lessons/projects/6.mdx [hunk 6]
```diff
@@ -0,0 +1,796 @@
...
+Once we have this values inside the file, we can source it to have this values
+as enviroment variables before using foundry's deploy tools.
+
+There are many ways to load these values into environment variables, depending
+on which shell you use (Bash, zsh, fish...). The next command works with most of
+bash-compatible shells:
+
+```bash
+export $(grep -v '^#' .env | xargs)
+```
```
Comment (georgemac510) [inline pages/lessons/projects/6.mdx:662]: Awesome! Some learners might be used to installing dotenv and adding the code for its use. How easy is that!

File: pages/lessons/projects/6.mdx (hunk 1 above)
Comment (7i7o) [reply pages/lessons/projects/6.mdx:36]: Already working on commiting all these corrections John! Thanks!

File: pages/lessons/projects/6.mdx [hunk 7]
```diff
@@ -0,0 +1,796 @@
...
+cd d_d_academy
+```
+
+You can use whatever tools you feel confident or accostumed to.
+For this example we will be creating a Foundry project, but feel free to use
+Hardhat or Truffle if that's your favourite flavor.
+The main focus of Foundry is that you don't need to use Javascript at all if you
+don't want to.
+All the tools are CLI commands. For tests and scripts, you only need Solidity!
+
```
[the remark this answers is not in the document]
Comment (7i7o) [reply pages/lessons/projects/6.mdx:138]: This one is a great idea! I would love to see a comparison of its strengths, weaknesses, performance, etc.

File: pages/lessons/projects/6.mdx (hunk 3 above)
Comment (7i7o) [reply pages/lessons/projects/6.mdx:122]: Yes. This is also worth mentioning!

Comment (7i7o) [reply pages/lessons/projects/6.mdx:122]: Well, this is the interesting part of using foundry... You don't need *Node.js* **at all** to do this lesson.
Do you recall using `npm`, `node`, `npx` or `yarn` commands during the lesson?

I'm gonna let that sink in... 😉

File: pages/lessons/projects/6.mdx (hunk 4 above)
Comment (7i7o) [reply pages/lessons/projects/6.mdx:286]: I prefer to keep instructions clean, and if there's someone with more advanced terminal knowledge, they can save as much time as they want.
I was trying to focus more on the Solidity content than on teaching how to use the terminal.

File: pages/lessons/projects/6.mdx (hunk 6 above)
Comment (7i7o) [reply pages/lessons/projects/6.mdx:662]: Yeah, I don't know if I need to clarify that this gets wiped when you end your terminal session.

**Comment by georgemac510** [comment]
Ahhh, good point. We didn’t touch Node. It doesn’t matter. 👍

File: pages/lessons/projects/6.mdx (hunk 6 above)
Comment (7i7o) [reply pages/lessons/projects/6.mdx:662]: I'm adding this so people know that they need to run this again if they close the terminal

**Comment by 7i7o** [comment]
@georgemac510 @briangershon Do you guys think I should move from Goerli to Sepolia because of Goerli's end of support next year?

**Comment by georgemac510** [comment]
Great point @7i7o We can avoid the redeploying and reconfig of tooling that happened after the Merge with Rinkeby, Ropsten and Kovan. The mining at https://sepolia-faucet.pk910.de is currently pretty generous. Nice heads up Tito!

**Review by elPiablo — APPROVED** [review]
Hey @7i7o I just added my details. It seems we're good to go!

**Comment by elPiablo** [comment]
@wolovim 
I still need to add the new screenshot images with the new contract on Sepolia. I made the screenshots, but didn't have time right now to upload them.

**Review by elPiablo — APPROVED** [review]
hey @7i7o could you give this a quick once over and merge it? or @wolovim if you get here first 😆 
## changes made:
- I just redeployed the contract on Sepolia.
- Changed all references in the input and output code from Goerli to Sepolia
- And any references to those in the lesson body from Goerli to Sepolia 
- Also changed the four Etherscan screenshots to reflect the new contract details

**Review by 7i7o — COMMENTED** [review]
Everything's good on my side @elPiablo 
I can't "Approve" because it's my PR

**Pull request merged**